app/helpers/application_helper.rb

```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+
+  def select_type(skill_type)
+    @character.skills.order(:name).select{ |type_skill| type_skill.type_of_skill == skill_type }
```

Comment:
Thank you, thought I tried to put the scope in character but for some reason it crashed, so I changed it to the Skill model and it worked, like this `scope :by_skill, ->(skill_name) { where(type_of_skill: skill_name) }`